src/MultiDocumenter.jl

```diff
@@ -188,6 +188,17 @@ function maybe_clone(docs::Vector{MultiDocRef})
             run(
                 `$(git()) clone --depth 1 $(doc.giturl) --branch $(doc.branch) --single-branch $(doc.upstream)`,
             )
+        else
+            @info "Updating existing clone at $(doc.upstream)"
+            run(`$(git()) -C $(doc.upstream) fetch origin $(doc.branch)`)
+            current_branch = strip(read(`$(git()) -C $(doc.upstream) rev-parse --abbrev-ref HEAD`, String))
+            if !isempty(current_branch) && (current_branch != doc.branch)
+                @info "Changing branch: `$current_branch` -> `$(doc.branch)`"
+                run(`$(git()) -C $(doc.upstream) checkout $(doc.branch)`)
+            elseif isempty(current_branch)
+                @error "current_branch was empty"
+            end
+            run(`$(git()) -C $(doc.upstream) reset --hard origin/$(doc.branch)`)
```

Comment:
I am also not super comfortable with the `reset --hard` here. Especially if `doc.upstream` is pointing into a subdirectory of a real directory. Actually, instead of `-C`, should probably use `--git-dir`..

Reply:
Is there ever any harm in hard-resetting a remote branch?

Reply:
I think if everything is normal, then it doesn't do anything wrong. But the cases I'm concerned about:

1. Something has gone horribly wrong, and `doc.upstream` is not actually a separate Git repo anymore, and then this command nukes changes on your main repository.
2. The user has actually gone into the cloned directories and modified stuff, and then we nuke their updates.

So I am wondering if something like a `git merge --ff-only` would be the better option here. I didn't want to do a simple `git pull`, since that, I believe, would fetch all branches.

Reply:
What I meant is that `reset --hard origin/mybranch` can't ever nuke any local changes, because those are in `mybranch` instead.

Reply:
I might be mistaken, but I believe `reset --hard` will change the commit your _current_ branch points to `origina/mybranch` (any commit not on other branches/repos will become lost-ish), and it will also nuke all uncommitted working tree changes.

Reply:
Oh, doh! Yeah, I just misread the command here :P

Ok, so how about we just checkout `remote/mybranch` for the build?